packages/x-date-pickers/src/internals/components/wrappers/MobileWrapper.tsx

```diff
@@ -1,56 +1,62 @@
 import * as React from 'react';
 import { WrapperVariantContext } from './WrapperVariantContext';
-import { PickersModalDialog, ExportedPickerModalProps } from '../PickersModalDialog';
+import {
+  PickersModalDialog,
+  ExportedPickerModalProps,
+  PickersModalDialogSlotsComponent,
+  PickersModalDialogSlotsComponentsProps,
+} from '../PickersModalDialog';
 import { DateInputPropsLike } from './WrapperProps';
 import { PickerStateWrapperProps } from '../../hooks/usePickerState';
+import { DateInputSlotsComponent } from '../PureDateInput';
 
 export interface MobileWrapperProps extends ExportedPickerModalProps {
   children?: React.ReactNode;
 }
 
+export interface MobileWrapperSlotsComponent
+  extends PickersModalDialogSlotsComponent,
+    DateInputSlotsComponent {}
+
+export interface MobileWrapperSlotsComponentProps extends PickersModalDialogSlotsComponentsProps {}
+
 export interface InternalMobileWrapperProps extends MobileWrapperProps, PickerStateWrapperProps {
   DateInputProps: DateInputPropsLike & { ref?: React.Ref<HTMLDivElement> };
   PureDateInputComponent: React.JSXElementConstructor<DateInputPropsLike>;
+  components?: Partial<MobileWrapperSlotsComponent>;
+  componentsProps?: Partial<MobileWrapperSlotsComponentProps>;
 }
 
 export function MobileWrapper(props: InternalMobileWrapperProps) {
   const {
-    cancelText,
     children,
-    clearable,
-    clearText,
     DateInputProps,
     DialogProps,
-    okText,
     onAccept,
     onClear,
     onDismiss,
     onCancel,
     onSetToday,
     open,
     PureDateInputComponent,
-    showTodayButton,
-    todayText,
+    components,
+    componentsProps,
     ...other
   } = props;
 
   return (
     <WrapperVariantContext.Provider value="mobile">
-      <PureDateInputComponent {...other} {...DateInputProps} />
+      <PureDateInputComponent components={components} {...other} {...DateInputProps} />
```

Comment:
We don't pass the `componentsProps` here ?
Even if none are used right now, it's maybe better to pass it to make sure we don't forget it when adding a new slot props

Reply:
That's a bit tricky. The components are already in `DateInputProps`

Maybe it will be easier to maintain if I move it from `DateInputProps` to the one you propose. But for now, I prefer to do this correction in a dedicated PR. The reason of the `DateInputProps` is still unclear to me